application_store_api/namespace_fund/data_store/store.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+import json
+import uuid
+
+from application_store_api.namespace_fund.data_store.data import (
+    funding_service_design_application,
+)
+from application_store_api.namespace_fund.data_store.data import (
+    initial_fund_store_state,
+)
+from dateutil import parser as date_parser
+from dateutil.tz import UTC
+from slugify import slugify
+
+"""
+Data Access Object
+A data access object (DAO) is a pattern that provides an abstract interface
+to some type of database or other persistence mechanism.
+"""
+
+
+class application_data_access_object(object):
+    def __init__(self):
+        self.counter = 0
+        self.funds = initial_fund_store_state
+
+    def get_funds(self):
+        # return all funds (replace the date object with a string to send)
+        return json.loads(json.dumps(self.funds, default=str))
+
+    def create_application(self, application):
+        application_fund_name = slugify(application["name"])
+        application_id = str(uuid.uuid4())  # cant be uuid in restx handler
+        current_time = datetime.datetime.now(datetime.timezone.utc)
+        application["date_submitted"] = current_time
+        application["id"] = application_id
+
+        if application_fund_name not in self.funds:
+            # create a new fund entry
+            self.funds[application_fund_name] = []
+
+        # place application within the fund
+        self.funds[application_fund_name].append(application)
+        return application
+
+    def get_applications_for_fund(
+        self, fund_name, datetime_start, datetime_end
+    ):
+        try:
+            fund_data = self.funds[fund_name]
+            applications_within_period = []
+            if datetime_start and datetime_end:
+                # convert string dates into datetimes
+                start = date_parser.parse(datetime_start).astimezone(UTC)
+                end = date_parser.parse(datetime_end).astimezone(UTC)
+
+                # compare period limits against application dates within fund
+                for application in self.funds[fund_name]:
+                    if (
+                        start
+                        < application["date_submitted"].astimezone(UTC)
+                        < end
+                    ):
+                        applications_within_period.append(application)
+                return json.loads(
+                    json.dumps(applications_within_period, default=str)
+                )
+            else:
+                return json.loads(json.dumps(fund_data, default=str))
+        except Exception:
```

Comment:
Lovely comments and variable names - very clear. The future generations of devs will be glad ❤️ 

In regards to the except Exception - 😵‍💫 😵‍💫 - while it is better to ask for forgiveness rather then permission please make sure you state what is being forgiven...IE instead of Exception try to find a specific error that relates to this...see https://docs.python.org/3/library/exceptions.html ...maybe some sort of NotFound exception.

I challenge you to find a solution which doesn't depend on using a try-except block...maybe we can use an if instead...

Reply:
@SentientHousePlant @Adam-W1 is this reoslved fellas?

Reply:
not yet - working on this one this morning